sentence_transformers/sparse_encoder/SparseEncoder.py

```diff
@@ -576,17 +576,18 @@ def encode(
             features = batch_to_device(features, self.device)
             features.update(extra_features)
 
-            with torch.no_grad():
+            with torch.inference_mode():
                 embeddings = self.forward(features, **kwargs)["sentence_embedding"].detach()
 
                 if max_active_dims:
                     embeddings = select_max_active_dims(embeddings, max_active_dims=max_active_dims)
-                if convert_to_sparse_tensor:
-                    embeddings = embeddings.to_sparse()
-                if save_to_cpu:
-                    embeddings = embeddings.cpu()
 
-                all_embeddings.extend(embeddings)
+            if convert_to_sparse_tensor:
+                embeddings = embeddings.to_sparse()
+            if save_to_cpu:
+                embeddings = embeddings.cpu()
+
+            all_embeddings.extend(embeddings)
```

Comment:
Due to what seems to be a bug in torch, I get issues with `coalesce` over the sparse embeddings if the `to_sparse` call happens inside of the inference-mode, so the inference-mode doesn't include it. The select_max_active_dims was required to be in the inference-mode, I get an error otherwise.

P.s. inference_mode also implies no_grad.